src/site/stages/build/process-cms-exports/transformers/helpers.js

```diff
@@ -76,6 +76,22 @@ function getDrupalValue(arr) {
   return null;
 }
 
+/**
+ * A very specific helper function that expects to receive an
+ * array with one item which is an object with a single `processed` property
+ *
+ */
+function getDrupalProcessed(arr) {
+  if (arr.length === 0) return null;
+  if (arr.length === 1)
+    return typeof arr[0].value === 'string'
+      ? unescapeUnicode(arr[0].processed)
+      : arr[0].value;
```

Comment:
Should this be checking `arr[0].processed` instead? And returning unescaped `arr[0].processed` when it's not a Unicode-escaped string?

I think the comment saying "object with a single `processed` property" might be inaccurate; if this is like the field I'm thinking of, I believe there are also `value` and `format` properties.

I don't know if it's really necessary to refactor now given the potential of eventually moving away from the transformer workflow, but here's a thought. In the vein of what Michael is suggesting, and in combination with [this Slack thread](https://dsva.slack.com/archives/C01512KN35G/p1602178081304700), I wonder if it's worth modiying `getDrupalValue` to prefer using the `processed` field if it exists and possibly if `format` is `'rich_text'`.

Reply:
Up to now, most of the properties use this structure:
```
"title": [
        {
            "value": "Pittsburgh VA Medical Center-University Drive"
        }
    ],
```
Just `value`, sometimes `format` but always ignored

However, `field_body` uses 3 properties

```
"field_body": [
        {
            "value": "<h3>Location and contact information<\/h3>\r\n\r\n<p class=\"va-address-block\"><strong>Primary Care Services<\/strong><br \/>\r\nBuilding 1<br \/>\r\n1st Floor, Annex Area<br \/>\r\n<a data-entity-substitution=\"canonical\" data-entity-type=\"node\" data-entity-uuid=\"71ffee52-a47f-4909-ad7e-21ece5728be3\" href=\"\/node\/783\",
            "format": "rich_text",
            "processed": "<h3>Location and contact information<\/h3>\n\n<p class=\"va-address-block\"><strong>Primary Care Services<\/strong><br \/>\nBuilding 1<br \/>\n1st Floor, Annex Area<br \/><a data-entity-substitution=\"canonical\" data-entity-type=\"node\" data-entity-uuid=\"71ffee52-a47f-4909-ad7e-21ece5728be3\" href=\"\/pittsburgh-health-care\/university-drive-campus-map\""
        }
    ],
```

In this case, it has 3 properties and we have been using `getDrupalValue` which is incorrect. I thought about using the existing `getDrupalValue` as @mpelzsherman mentioned and passing an argument but that would be a lot of refactoring.

Maybe having an argument default value like `property = "value"` and only pass the argument when the `processed` field is needed instead. And an `if statement` for getting either `value` or `processed`.

I thought it was easy just create a separate function.

What do you think would be better?

Reply:
I'll try to refactor it

Reply:
I would opt for the easier solution if it ends up being a big refactor, since its value is lost once the transformation procecss becomes deprecated.